tests/control/PlanarArmControllerTest.cpp

```diff
@@ -29,7 +29,8 @@ void assertApprox(const Eigen::Vector2d& p1, const Eigen::Vector2d& p2, double d
 TEST_CASE("Test Planar Arm Controller", "[control][planararmcontroller]") {
 	navtypes::Vectord<2> vec({0, 0});
 	kinematics::PlanarArmKinematics<2> kin_obj(vec, vec, vec, 0.0, 0);
-	PlanarArmController<2> foo({0, 0}, kin_obj, Constants::arm::SAFETY_FACTOR);
+	PlanarArmController<2> foo(kin_obj, Constants::arm::SAFETY_FACTOR);
+	REQUIRE(foo.tryInitController({0, 0}));
```

Comment:
Also this test case doesn't do anything (it was initially added to force the template to be instantiated, otherwise it wouldn't be compiled) and since it initializes the segLens as 0, it will break a lot of the math, as you can see in the failing test case right now. I'd say just get rid of this test case, and then add some `CHECK_F`s in the constructor to make sure all the `segLens` are strictly positive.